packages/design-system/src/components/button.stories.tsx

```diff
@@ -1,10 +1,94 @@
-import type { ComponentStory } from "@storybook/react";
-import { Button } from "./button";
+import { type ComponentProps } from "react";
+import { MenuIcon, CrossIcon, TrashIcon } from "@webstudio-is/icons";
+import { Button as ButtonComponent } from "./button";
 
 export default {
-  component: Button,
+  title: "Library/Button",
 };
 
-export const Simple: ComponentStory<typeof Button> = () => {
-  return <Button>Simple</Button>;
+const iconsMap = {
+  undefined: undefined,
+  "<MenuIcon>": <MenuIcon />,
+  "<CrossIcon>": <CrossIcon />,
+  "<TrashIcon>": <TrashIcon />,
+} as const;
+
+const Section = ({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) => (
+  <>
+    <h3 style={{ fontFamily: "sans-serif" }}>{title}</h3>
+    <div style={{ display: "flex", gap: 12 }}>{children}</div>
+  </>
+);
+
+export const Button = ({
+  icon,
+  ...rest
+}: Omit<ComponentProps<typeof ButtonComponent>, "icon"> & {
+  icon?: keyof typeof iconsMap;
+}) => (
+  <>
+    <Section title="Configurable">
+      <ButtonComponent icon={icon && iconsMap[icon]} {...rest} />
+    </Section>
+
+    <Section title="Variants">
+      {variants.map((variant) => (
+        <ButtonComponent variant={variant} key={variant}>
+          {variant}
+        </ButtonComponent>
+      ))}
+    </Section>
+
+    <Section title="With Icon">
+      <ButtonComponent icon={<TrashIcon />}>Button</ButtonComponent>
+      <ButtonComponent icon={<TrashIcon />} iconPosition="right">
+        Button
+      </ButtonComponent>
+      <ButtonComponent icon={<TrashIcon />} />
+    </Section>
+
+    <Section title="Disabled">
+      <ButtonComponent disabled>Button</ButtonComponent>
+      <ButtonComponent disabled icon={<TrashIcon />}>
+        Button
+      </ButtonComponent>
+      <ButtonComponent disabled icon={<TrashIcon />} />
```

Comment:
can pending and disabled be at the same time? isn't it impossible state made possible?

Reply:
As it implemented currently `pending=true disabled=true` is equivalent to `pending=true disabled=false`, so although it's a weird state there's no problem.

Will keep as is, because an API that will not allow this state will be more confusing.